File: cs/MultipleDispatch/MultipleDispatchTests/PrettyPrinter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Collections;
using MultipleDispatch;

namespace MultipleDispatchTests
{
    class PrettyPrinter
    {
        public string Print(string str)
        {
            return String.Format("STRING({0})", str);
        }

        public string Print(object val, string comment)
        {
            return Multimethod.Call<string>(this, "Print", val) + " // " + comment;
        }

        public string Print(float value)
        {
            return String.Format("FLOAT({0})", value);
        }

        public string Print(double value) 
        {
            return String.Format("DOUBLE({0})", value);
        }

        public string Print(int value)
        {
            return String.Format("INT({0})", value);
        }

        public string Print(object value)
        {
            return "???";
        }

        public string Print(IEnumerable collection)
        {
            string result = "[";

            bool first = true;
            foreach (var obj in collection)
            {
                if (!first)
                    result += ", ";

                result += Multimethod.Call<string>(this, "Print", obj);

                if (first) first = false;
            }

            return result + "]";
        }
    }
}

File: src/java/MultipleDispatchTest/src/multipledispatch/PrettyPrinter.java
package multipledispatch;

/**
 * A simple pretty printer used to test the Multimethod.
 * 
 * @author "Maciej Pacula (maciej.pacula@gmail.com)"
 *
 */
public class PrettyPrinter {
	public String print(String str)
    {
        return String.format("STRING(%s)", str);
    }

    public String print(Object val, String comment)
    {
        return Multimethod.call(String.class, this, "print", val) + " // " + comment;
    }

    public String print(float value)
    {
        return String.format("FLOAT(%f)", value);
    }

    public String print(double value) 
    {
        return String.format("DOUBLE(%f)", value);
    }

    public String print(int value)
    {
        return String.format("INT(%d)", value);
    }

    public String print(Object value)
    {
        return "???";
    }

    public String print(Object[] array)
    {
        String result = "[";

        boolean first = true;
        for (Object obj : array)
        {
            if (!first)
                result += ", ";

            result += Multimethod.call(String.class, this, "print", obj);

            if (first) first = false;
        }

        return result + "]";
    }
}
